src/dataset.cpp

```diff
@@ -61,16 +61,30 @@ VariableSlice Dataset::operator()(const Tag tag, const std::string &name) & {
 }
 
 void Dataset::insert(Variable variable) {
-  for (const auto &item : m_variables)
-    if (item.tag() == variable.tag() && item.name() == variable.name())
-      throw std::runtime_error(
-          "Attempt to insert variable with duplicate tag and name.");
   // TODO special handling for special variables types like
   // Data::Histogram (either prevent adding, or extract into underlying
   // variables).
-  mergeDimensions(variable.dimensions(),
-                  coordDimension[variable.tag().value()]);
-  m_variables.push_back(std::move(variable));
+  if (contains(variable.tag(), variable.name())) {
```

Comment:
Seems like this should be broken up into a few sub functions, maybe something for later clean-up?

Reply:
Yes, I started that, but then aborted, since it led to too many changes at this point. For example, we have `Dataset::find`, which throws, but returning `-1` would be a more standard behavior. A number of internals and API functions should be refactored.